src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewManfiest.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    public static class CompiledViewManfiest
+    {
+        public static readonly string PrecompiledViewsAssemblySuffix = ".PrecompiledViews";
+
+        public static TManifest LoadManifest<TManifest>(AssemblyPart assemblyPart, string typeName)
```

Comment:
The page and view manifest are slightly different (pages have the route template and soon, the name embedded). Even if we switch to attributes, we would need two separate types.